# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/base/threading/platform_thread_linux.cpp
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/base/threading/platform_thread_linux.cpp
--- a/security/sandbox/chromium-shim/base/threading/platform_thread_linux.cpp	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium-shim/base/threading/platform_thread_linux.cpp	Wed Jun 12 11:10:48 2019 +0100
@@ -28,21 +28,26 @@
     {ThreadPriority::REALTIME_AUDIO, -10},
 };
 
+
+Optional<bool> CanIncreaseCurrentThreadPriorityForPlatform(
+    ThreadPriority priority) {
+  MOZ_CRASH();
+}
+
 bool SetCurrentThreadPriorityForPlatform(ThreadPriority priority) {
   MOZ_CRASH();
 }
 
-bool GetCurrentThreadPriorityForPlatform(ThreadPriority* priority) {
+Optional<ThreadPriority> GetCurrentThreadPriorityForPlatform() {
   int maybe_sched_rr = 0;
   struct sched_param maybe_realtime_prio = {0};
   if (pthread_getschedparam(pthread_self(), &maybe_sched_rr,
                             &maybe_realtime_prio) == 0 &&
       maybe_sched_rr == SCHED_RR &&
       maybe_realtime_prio.sched_priority == kRealTimePrio.sched_priority) {
-    *priority = ThreadPriority::REALTIME_AUDIO;
-    return true;
+    return base::make_optional(ThreadPriority::REALTIME_AUDIO);
   }
-  return false;
+  return base::nullopt;
 }
 
 }  // namespace internal